### PR TITLE
MM-12443 Migrate 'components/post_deleted_modal.jsx' and associated tests to TypeScript

### PR DESCRIPTION
**Pull request by brewsterbhg**
#### Summary
Migrate 'components/post_deleted_modal.jsx' and associated tests to TypeScript.

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/12443

### CODE REVIEW

**Comment by lieut-data** [comment]
@devinbinnie, I'd like to kick this back to you since it involves `package.json` changes with which I'm unfamiliar (want to make sure this is as expected).

File: components/post_deleted_modal.test.tsx [hunk 1]
```diff
@@ -59,8 +59,8 @@ describe('components/ChannelInfoModal', () => {
                 onHide={emptyFunction}
             />
         );
-
-        const shouldUpdate = wrapper.instance().shouldComponentUpdate({show: true});
+        const instance = wrapper.instance();
+        const shouldUpdate = instance.shouldComponentUpdate && instance.shouldComponentUpdate({show: true}, {}, null);
```
Comment (devinbinnie) [inline components/post_deleted_modal.test.tsx]: Can `shouldComponentUpdate` be null?

Comment (brewsterbhg) [reply components/post_deleted_modal.test.tsx]: It should never be null, but the type checking was failing and I wasn't sure of a more graceful solution for getting TS to play ball in this case (I've never had to call lifecycle methods from a test before)

Comment (devinbinnie) [reply components/post_deleted_modal.test.tsx]: You can use the `!.` operator to have the TypeScript compiler dismiss the null check. If it should never be null, the app should throw an error if it is.

Comment (brewsterbhg) [reply components/post_deleted_modal.test.tsx]: I saw that there was a lint rule against using the `!` operator, so I wasn't sure if that was something we were avoiding, but I can do it and just disable the rule for this line

Comment (devinbinnie) [reply components/post_deleted_modal.test.tsx]: I've removed that lint rule, you should be good to use it as long as your branch is sync'd with the latest master.

Comment (brewsterbhg) [reply components/post_deleted_modal.test.tsx]: Alright cool, thanks for letting me know. I've synced and made that update

File: components/post_deleted_modal.test.tsx [hunk 2]
```diff
@@ -59,8 +59,8 @@ describe('components/ChannelInfoModal', () => {
                 onHide={emptyFunction}
             />
         );
-
-        const shouldUpdate = wrapper.instance().shouldComponentUpdate({show: true});
+        const instance = wrapper.instance();
```
Comment (marianunez) [inline components/post_deleted_modal.test.tsx]: I believe this is no longer needed

File: package.json [hunk 3]
```diff
@@ -11,6 +11,7 @@
     "@types/jest": "24.0.17",
     "@types/node": "12.7.0",
     "@types/react": "16.8.24",
+    "@types/react-bootstrap": "0.32.20",
```
Comment (marianunez) [inline package.json]: Do we need to include this?

Comment (devinbinnie) [reply package.json]: No, it's already in the `devDependencies`. @brewsterbhg can we remove this?

**Comment by saturninoabril** [comment]
@brewsterbhg Please help sync to master so we can merge.

**Comment by brewsterbhg** [comment]
> @brewsterbhg Please help sync to master so we can merge.

synced!

**Pull request merged**